tensorflow_addons/activations/rrelu.py

```diff
@@ -44,19 +44,28 @@ def rrelu(
         training: `bool`, indicating whether the `call`
         is meant for training or inference.
         seed: `int`, this sets the operation-level seed.
+        gs: A `Generator`. Default value is tf.random.get_global_generator().
     Returns:
         result: A `Tensor`. Has the same type as `x`.
     """
     x = tf.convert_to_tensor(x)
+    lower = tf.cast(lower, x.dtype)
+    upper = tf.cast(upper, x.dtype)
+
     if training is None:
         training = tf.keras.backend.learning_phase()
         training = bool(tf.keras.backend.get_value(training))
 
     if training:
-        alpha = tf.random.uniform(
-            tf.shape(x), minval=lower, maxval=upper, dtype=x.dtype, seed=seed
-        )
+        if gs is None:
+            alpha = tf.random.uniform(
+                tf.shape(x), minval=lower, maxval=upper, dtype=x.dtype, seed=seed
+            )
+        else:
+            if seed is not None:
```

Comment:
In term of control flow, I believe something flatter would be easier to read. Something like:

```python
if gs is not None and seed is not None:
    raise ValueError(...)
elif gs is not None:
    gs.uniform(...)
else:
    tf.random.uniform(..., seed=seed)
```